lib/galaxy/web_stack/handlers.py

```diff
@@ -53,6 +54,33 @@ def add_handler(self, handler_id, tags):
             else:
                 self.handlers[tag] = [handler_id]
 
+    @staticmethod
+    def supports_returning(model: GalaxyModelMapping):
+        job_table = model.Job.table
+        stmt = job_table.update().where(job_table.c.id == -1).returning(job_table.c.id)
+        try:
+            model.session.execute(stmt)
+            return True
+        except Exception:
```

Comment:
Is there a more specific exception we can catch here and in `supports_skip_locked()` ?

Also, we could `log.debug()` a message.

Reply:
This would be hypothetical, since sqlalchemy forwards the underlying library exception, so would be different for psyopg2 / asyncpg / sqlite3. I'll think about what we can log here.

Reply:
fwiw these are the exception I've seen so far:
For psycopg2 + postgres 8 and the exception for with_for_update(skip_locked=True) is 

```
ProgrammingError: (psycopg2.errors.SyntaxError) syntax error at or near "SKIP"
LINE 3: WHERE job.id =  -1 FOR UPDATE SKIP LOCKED
```

for sqlite (sqlalchemy hasn't updated this yet, since the first release that supports it is very new) this is currently
```
CompileError: RETURNING is not supported by this dialect's statement compiler.
```

If I remove that explicit exception from sqlalchemy and replace it with the actual implementation it either works (sqlite > 3.35.0) or fails with
```
OperationalError: (sqlite3.OperationalError) near "RETURNING": syntax error
[SQL: UPDATE job SET update_time=? WHERE job.id = ? RETURNING job.id]
[parameters: ('2021-04-06 11:33:26.931462', -1)]
(Background on this error at: http://sqlalche.me/e/13/e3q8)
```

Reply:
Ah, I thought we would get a SQLAlchemy error. Then I guess a log message would be good enough.